target/linux/ath79/dts/qca9557_ruckus_r500.dts

```diff
@@ -0,0 +1,324 @@
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+	};
+
+	beamforming-2g-spi {
+		compatible = "spi-gpio";
+		mosi-gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		sck-gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <0>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		beamforming-2g-gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			registers-number = <1>;
+			spi-max-frequency = <24000000>;
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+	};
+
+	beamforming-5g-spi {
+		/*
+		 * ath10k driver doesn't expose GPIOs as of now,
+		 * so let's disable it for now. The shift register
+		 * for antenna pattern control is correctly initialized
+		 * for omnidirectional pattern correctly despite that,
+		 * there is just no possibility to control that from OpenWrt yet.
+		 */
+		status = "disabled";
+		compatible = "spi-gpio";
+		mosi-gpios = <&ath10k 15 GPIO_ACTIVE_HIGH>;
+		sck-gpios = <&ath10k 14 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <0>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		beamforming-5g-gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			registers-number = <1>;
+			spi-max-frequency = <24000000>;
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+	};
+
+	i2c: i2c {
+		compatible = "i2c-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		scl-gpios = <&gpio 19 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		sda-gpios = <&gpio 20 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+
+		tpm@20 {
+			compatible = "infineon,slb9645tt", "tcg,tpm-tis-i2c";
+			reg = <0x20>;
+			crc-checksum;
+		};
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		ruckus-himem@fff0000 {
+			/* Ruckus Himem area used to control
+			 * redundant boot image selection
+			 */
+			compatible = "nvmem-rmem";
+			reg = <0xfff0000 0x10000>;
+			no-map;
+		};
+	};
+};
+
+&pinmux {
+	clks_disable_pins: pinmux_clks_disable_pins {
+		pinctrl-single,bits = <0x40 0x0 0x80>;
+	};
+
+	enable_gpio_4: pinctrl_enable_gpio_4 {
+		pinctrl-single,bits = <0x4 0x0 0xff>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x0000000 0x100000>;
+				label = "u-boot";
+				read-only;
+			};
+
+			partition@100000 {
+				compatible = "u-boot,env";
+				reg = <0x0100000 0x40000>;
+				label = "u-boot-env";
+			};
+
+			board_data: partition@140000 {
+				reg = <0x0140000 0x80000>;
+				label = "board-data";
+				read-only;
+			};
+
+			partition@1c0000 {
+				reg = <0x01c0000 0x3e40000>;
+				compatible = "openwrt,uimage", "denx,uimage";
+				label = "firmware";
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+
+		qca,ar8327-initvals = <
```

Comment:
What was the policy when it came to qca8k here again?

Reply:
Right. qca8k is available as a module now. Though from what I remember @hauke was opposed to doing conversions to qca8k.

Reply:
Yes ath79 still uses swconfig by default.